common-files/test/httputils.test.mjs

```diff
@@ -0,0 +1,179 @@
+      } catch (err) {
+        next(err);
+      }
+    });
+
+    setupHttpDefaults(app_0, app_0 => {
+      app_0.use(routerApp0);
+    });
+
+    app_0.listen(APP_0_PORT);
+
+    // app#1
+    const routerApp1 = express.Router();
+
+    routerApp1.get('/test', async (req, res) => {
+      correlationIdReceivedApp1 = req.header('X-Correlation-Id');
+      res.sendStatus(200);
+    });
+
+    setupHttpDefaults(
+      app_1,
+      app_1 => {
+        app_1.use(routerApp1);
+      },
+      false,
+      false,
+    );
+
+    app_1.listen(APP_1_PORT);
+  });
+
+  it('Should create the healthcheck endpoint for app#0', function (done) {
+    chai
+      .request(app_0)
+      .get('/healthcheck')
+      .end((err, res) => {
+        res.should.have.status(200);
+        done();
+      });
+  });
+
+  it('Should not create the healthcheck endpoint for app#1', function (done) {
+    chai
+      .request(app_1)
+      .get('/healthcheck')
+      .end((err, res) => {
+        res.should.have.status(404);
+        done();
+      });
+  });
+
+  it('Should reply with CORS header', function (done) {
+    chai
+      .request(app_0)
+      .get('/healthcheck')
+      .end((err, res) => {
+        res.should.have.header('Access-Control-Allow-Origin');
+        done();
+      });
+  });
+
+  it('Shouldn`t reply with header `x-powered-by`', function (done) {
+    chai
+      .request(app_0)
+      .get('/healthcheck')
+      .end((err, res) => {
+        res.should.not.have.header('x-powered-by');
+        done();
+      });
+  });
+
+  it('Should create routes accordingly', function (done) {
+    chai
+      .request(app_0)
+      .get('/test/111')
+      .end((err, res) => {
+        res.should.have.status(200);
+        done();
+      });
+  });
+
+  it('Should reply with the correlationId header', function (done) {
+    chai
+      .request(app_0)
+      .get('/test/111')
+      .end((err, res) => {
+        res.should.have.header('X-Correlation-Id');
+        done();
+      });
+  });
+
+  it('Should receive HTTP status 400', function (done) {
+    chai
+      .request(app_0)
+      .get('/test/aaa')
+      .end((err, res) => {
+        res.should.have.status(400);
+        done();
+      });
+  });
+
+  it('Should receive HTTP status 404', function (done) {
+    chai
+      .request(app_0)
+      .get('/test/0')
+      .end((err, res) => {
+        res.should.have.status(404);
+        done();
+      });
+  });
+
+  it('Should receive HTTP status 500', function (done) {
+    chai
+      .request(app_0)
+      .get('/test/999')
+      .end((err, res) => {
+        res.should.have.status(500);
+        done();
+      });
+  });
+
+  it('Should pass the `correlationId` header from `app#0` to `app#1` for any HTTP request perfomed', function (done) {
+    chai
+      .request(app_0)
+      .get('/test/111')
+      .end((err, res) => {
+        res.should.have.header('X-Correlation-Id', correlationIdReceivedApp1);
+        done();
+      });
+  });
+});
```

Comment:
Very nice test! :boom:

Reply:
great stuff @israelboudoux so happy we're adding great tests like this